Docs: Create soc.md (#1325)
deepmodeling/abacus-develop: docs/advanced/scf/soc.md

## SOC Effects
### SOC 
`lspinorb` is used for control whether or not SOC(spin-orbit coupling) effects should be considered.

Both `basis_type=pw` and `basis_type=lcao` support `scf` and `nscf` calculation with SOC effects.

> Atomic forces and cell stresses can not calculated with SOC effects until now. 

Reviewer: In my humble opinion, "until now" means you can not do something before, but you can do it from now on. I guess this is not what you mean.
```suggestion
Atomic forces and cell stresses can not be calculated with SOC effects yet. 
```

### Pseudopotentials and Numerical Atomic Orbitals

> For Norm-Conserving pseudopotentials, there are difference between SOC version and non-SOC version.

Reviewer: ```suggestion
For Norm-Conserving pseudopotentials, there are differences between SOC version and non-SOC version.
```

Please check your pseudopotential files before calculating.
In `PP_HEADER` part, keyword `has_so=1` and `relativistic="full"` refer to SOC effects have been considered, 
which would lead to different `PP_NONLOCAL` and `PP_PSWFC` parts.

> Please be careful that `relativistic="full"` version can be used for SOC or non-SOC calculation, but `relativistic="scalar"` version only can be used for non-SOC calculation

Reviewer: ```suggestion
Please be careful that `relativistic="full"` version can be used for SOC or non-SOC calculation, but `relativistic="scalar"` version only can be used for non-SOC calculation.
```

> When full-relativistic pseudopotential is used for non-SOC calculation, ABACUS will automatic transform it to scalar-relativistic version.

Reviewer: ```suggestion
When full-relativistic pseudopotential is used for non-SOC calculation, ABACUS will automatically transform it to scalar-relativistic version.
```

> Numerical atomic orbitals in ABACUS are unrelated with spin, same orbital file can be used for SOC and non-SOC calculation.

Reviewer: ```suggestion
Numerical atomic orbitals in ABACUS are unrelated with spin, and same orbital file can be used for SOC and non-SOC calculation.
```
